packages/framework-provider-local/src/library/read-model-adapter.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  BoosterConfig,
+  Filter,
+  Logger,
+  ReadModelEnvelope,
+  ReadModelInterface,
+  UUID,
+} from '@boostercloud/framework-types'
+import { ReadModelRegistry } from '../services/read-model-registry'
+
+export async function rawReadModelEventsToEnvelopes(
+  config: BoosterConfig,
+  logger: Logger,
+  rawEvents: Array<unknown>
+): Promise<Array<ReadModelEnvelope>> {
+  return rawEvents.map((event) => event as ReadModelEnvelope)
+}
+
+export async function fetchReadModel(
+  db: ReadModelRegistry,
+  _config: BoosterConfig,
+  logger: Logger,
+  readModelName: string,
+  readModelID: UUID
+): Promise<ReadModelInterface> {
+  const response = await db.query({ typeName: readModelName, value: { id: readModelID } })
+  const item = response[0]
+  if (!item) {
+    console.log(`[ReadModelAdapter#fetchReadModel] Failed to fetch read model ${readModelName} with ID ${readModelID}`)
+  } else {
+    logger.debug(
+      `[ReadModelAdapter#fetchReadModel] Loaded read model ${readModelName} with ID ${readModelID} with result:`,
+      item.value
+    )
+  }
+  return item?.value
+}
+
+export async function storeReadModel(
+  db: ReadModelRegistry,
+  _config: BoosterConfig,
+  logger: Logger,
+  readModelName: string,
+  readModel: ReadModelInterface
+): Promise<void> {
+  await db.store({ typeName: readModelName, value: readModel })
+  logger.debug('[ReadModelAdapter#storeReadModel] Read model stored')
+}
+
+export async function searchReadModel(
+  db: ReadModelRegistry,
+  _config: BoosterConfig,
+  logger: Logger,
+  readModelName: string,
+  filters: Record<string, Filter<unknown>>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Promise<Array<any>> {
+  logger.debug('[ReadModelAdapter#searchReadModel] Arrived these filters: ', filters)
```

Comment:
What do you think about failing if there are filters defined saying "not yet supported"?

Also, could you please create a ticket to implement the filtering?

Reply:
Actually there's already a ticket for this https://github.com/boostercloud/booster/issues/318

Reply:
Fixed in https://github.com/boostercloud/booster/pull/473/commits/f9fb9521fa7b31273c089704b9fea1e7e10347b7